### PR TITLE
Add helm option to set enableAutoMtls

### PR DESCRIPTION
**Pull request by diemtvu**
This PR adds enable_auto_mtls to configmap. This flag was introduced in https://github.com/istio/api/pull/1028.

Issue: https://github.com/istio/istio/issues/16586

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (rshriram) [inline install/kubernetes/helm/istio/templates/configmap.yaml]: mention that by default it is true

File: install/kubernetes/helm/istio/values.yaml
Comment (rshriram) [inline install/kubernetes/helm/istio/values.yaml]: why use a new name here? I suggest using automaticMtls here. And please set it to true. There is no reason to require people to know that this exists and configure it to true.

Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: Change to `automatic`, as it's already under mtls group. The reason I add `client` in the previous name as this mostly affect client side.

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: Done

File: install/kubernetes/helm/istio/values.yaml
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: i thought we wanted to dark launch this first, no?

I agree with @rshriram on the naming, i'd prefer to be autoMTLS or autoMtls.  Having client in the name can trigger more questions than necessary.

**Comment by linsun** [comment]
cc @GregHanson

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: Revert. We want default to `false` before all implementation complete and well tested.

File: install/kubernetes/helm/istio/values.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: Changed to `automatic`

**Comment by diemtvu** [comment]
Ping @linsun @rshriram 
Note this change have no impact yet, until some implementation (e.g #16657) submitted. However, it's needed for e2e test, so I want to have this separated PR to keep it clean.

**Review by rshriram — APPROVED** [review]

**Comment by diemtvu** [comment]
@costinm @ayj @howardjohn , can I have maintainer approval for this PR? Thanks

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (incfly) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: I think we can recommend or shed some light on this will be planed as default behavior in future releases. So it motivate customers better to opt-in earlier, and have a better preparation for the migration.

Just a thought.

File: install/kubernetes/helm/istio/values.yaml
Comment (ayj) [inline install/kubernetes/helm/istio/values.yaml]: What does appropriately mean in this context?

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: IMO, it's better to do that when we ready to make it default. But if you have strong opinion about it, I can add it.

**Review by diemtvu — COMMENTED** [review]
ping

File: install/kubernetes/helm/istio/values.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: Done

File: install/kubernetes/helm/istio/values.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: I was using autoClientConfig but was suggested to keep it closed to the mesh field (which name `enableAutoMtls`). I do agree `automaticClientMtsl` (or may be just `automaticClient` as it's already under mtls group) is more semantic accurate. Wdyt? + @rshriram + @incfly

Comment (ayj) [reply install/kubernetes/helm/istio/values.yaml]: If `automaticClientMTLS` is a more accurate API we should rename the field in MeshConfig. `enableAutoMtls` is only in master and hasn't shipped to end users yet.

Comment (incfly) [reply install/kubernetes/helm/istio/values.yaml]: What do we want to call this feature in our docs/branding? I think at least for that `auto mtls` is a better terminolgy, even though the delta we're doing for 1.4 mostly involves client. The server side permissive mode work is done.

@linsun @rshriram thoughts?

Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: It's currently `automatic`. I'll change to `auto` as it seems in favor at the moment.

File: install/kubernetes/helm/istio/templates/configmap.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: `enableAutoMtls` is the mesh config (API). We can change if everyone agree, but need to be done in API first.

File: install/kubernetes/helm/istio/values.yaml
Comment (diemtvu) [reply install/kubernetes/helm/istio/values.yaml]: Enforcement is on client side, but DR also has target (host) that is server side. I always have trouble with explaining this DR anyway. Do you have suggestion?

**Comment by diemtvu** [comment]
Change to `auto`. Please take another look.

**Comment by GregHanson** [comment]
@diemtvu since you have made changes to the install/ dir, make sure you submit a corresponding PR to `install` repo
cc @sdake

**Comment by diemtvu** [comment]
ping

**Review by ayj — APPROVED** [review]

**Pull request merged**
